### ISSUE TITLE
unnecessary-semicolon false positive for f-string text with an explicit continuation on Python 3.12+

### ISSUE THREAD

**Issue by joseoteromato** [pylint-dev/pylint#11444]
### Bug description

On Python 3.12+, PEP 701 tokenizes `;` here as `FSTRING_MIDDLE`, where it is string content. Pylint reports it as an unnecessary statement terminator when the following implicitly concatenated string is on an explicit continuation line. `demo.py`:

```python
b = "value"
a = f"{b};" \
""
```

### Command used

```shell
pylint --disable=all --enable=unnecessary-semicolon --rcfile=/dev/null demo.py
```

### Pylint output

```python
************* Module demo
demo.py:2:0: W0301: Unnecessary semicolon (unnecessary-semicolon)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```

### Expected behavior

No `unnecessary-semicolon` warning. The semicolon is part of the f-string literal, not a Python statement terminator.

### Pylint version

```shell
pylint 4.0.8
astroid 4.0.4
Python 3.14.7 (main, Sep  1 2026, 14:18:09) [Clang 22.1.3 ]

Confirmed also for python3.12 and older versions of pylint such as 2.13.7. Pylint 4.0.8 does not report it on Python 3.11.
```
